File: src/Weapon.java
import java.util.List;
import java.util.ArrayList;

public class Weapon{
    private int magCapacity;
    private int totalAmmo;
    private int damage;
    private List<Shot> shots;

    public Weapon(){
        this.damage = 25;
        this.magCapacity = 30;
        this.totalAmmo = 90;

        shots = new ArrayList<Shot>();
    }
    //Método para atirar
    public void shoot(Player p){
        if(magCapacity > 0){
            this.shots.add(new Shot(((p.getX() + p.getW())), (p.getY()+13), damage, p.getDirection()));
            this.magCapacity -= 1;
        }
    }
    //Método para recarregar
    public void reload(){
        if (totalAmmo > 0) {
            setMagCapacity(30);
            this.totalAmmo -= 30;
            System.out.println(this.totalAmmo);
        }
    }
    //Método para adicionar mais munição no inventário
    public void addAmmo(int ammo){
        this.totalAmmo += ammo;
        System.out.println(this.totalAmmo);
    }
    //Getter para tiros
    public void setMagCapacity(int magCapacity) {
        this.magCapacity = magCapacity;
    }
    //Setter para tiros
    public List<Shot> getShots(){
        return shots;
    }
    //Getter para objeto Weapon
    public Weapon getWeapon(){
        return this;
    }
    public int getTotalAmmo() {
        return totalAmmo;
    }
}

File: scr_js/Weapon.js
class Weapon extends Entity{
    constructor(animations, pst = {}){
        super(animations);

        this.life.bar = 0;

        this.position = {
            x: pst.x,
            y: pst.y
        };
        this.mag = {
            standart: 30,
            total: 120
        }
        this.damage = 25;

        this.shots = [];
        this.shooting = false;
    }

    move(pst = {}, d){
        this.position = {
            x: pst.x+7,
            y: pst.y+7
        };
        this.direction = d;
    }

    //Método para atirar
    shoot(){
        if(this.mag.standart > 0){
            this.shots.push(new Shot((this.position.x + this.size.width), (this.position.y+5), this.direction, 
                                {standart: new Sprite('./img/shot.png', 1, 0),
                                 standartL: new Sprite('./img/shot-l.png', 1, 0),
                                }));
            this.mag.standart -= 1;
        }
    }
    //Método para recarregar
    reload(){
        if (this.mag.total > 0 && this.mag.standart != 30) {
            this.mag.standart += 30;
            this.mag.total -= 30;
            console.log(this.mag.total);
        }
    }
    //Método para adicionar mais munição no inventário
    addAmmo(ammo){
        this.mag.total += ammo;
        console.log(this.mag.total);
    }
}